include/powerloader/curl.hpp

```diff
@@ -35,17 +36,25 @@ namespace powerloader
         bool m_serious;
     };
 
-    struct POWERLOADER_API Response
+    struct POWERLOADER_API BaseResponse
     {
-        std::map<std::string, std::string> header;
-        mutable std::stringstream content;
+        std::map<std::string, std::string> headers;
```

Comment:
While we're at it, I would have used something like `boost::container::flat_map` which is a vector of pair with a map interface. As an optimisation though, so not necessary to do now.

Reply:
Ah that was just an example, this kind of structure is pretty common (and proposed to standardisation) but yeah.
My worry is mainly that `std::map` search is terrible, in particular with small data.
Anyway you can ignore this comment, it was just a note to myself I guess.